src/core/indexer/tables.rs

```diff
@@ -0,0 +1,454 @@
+
+        s = sizage(Codex::Ed25519_Big_Sig.code()).unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 4);
+        assert_eq!(s.os, 2);
+        assert_eq!(s.fs, 92);
+        assert_eq!(s.ls, 0);
+
+        s = sizage(Codex::Ed25519_Big_Crt_Sig.code()).unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 4);
+        assert_eq!(s.os, 2);
+        assert_eq!(s.fs, 92);
+        assert_eq!(s.ls, 0);
+
+        s = sizage(Codex::ECDSA_256k1_Big_Sig.code()).unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 4);
+        assert_eq!(s.os, 2);
+        assert_eq!(s.fs, 92);
+        assert_eq!(s.ls, 0);
+
+        s = sizage(Codex::ECDSA_256k1_Big_Crt_Sig.code()).unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 4);
+        assert_eq!(s.os, 2);
+        assert_eq!(s.fs, 92);
+        assert_eq!(s.ls, 0);
+
+        s = sizage(Codex::Ed448_Big_Sig.code()).unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 6);
+        assert_eq!(s.os, 3);
+        assert_eq!(s.fs, 160);
+        assert_eq!(s.ls, 0);
+
+        s = sizage(Codex::Ed448_Big_Crt_Sig.code()).unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 6);
+        assert_eq!(s.os, 3);
+        assert_eq!(s.fs, 160);
+        assert_eq!(s.ls, 0);
+
+        s = sizage(Codex::TBD0.code()).unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 2);
+        assert_eq!(s.os, 0);
+        assert_eq!(s.fs, u32::MAX);
+        assert_eq!(s.ls, 0);
+
+        s = sizage(Codex::TBD1.code()).unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 2);
+        assert_eq!(s.os, 1);
+        assert_eq!(s.fs, 76);
+        assert_eq!(s.ls, 1);
+
+        s = sizage(Codex::TBD4.code()).unwrap();
+        assert_eq!(s.hs, 2);
+        assert_eq!(s.ss, 6);
+        assert_eq!(s.os, 3);
+        assert_eq!(s.fs, 80);
+        assert_eq!(s.ls, 1);
+    }
+
+    #[test]
+    #[should_panic(expected = "UnknownSizage(\"z\")")]
+    fn test_unkown_size() {
+        sizage("z").unwrap();
+    }
+
+    #[test]
+    fn test_hardage() {
+        assert_eq!(hardage('A').unwrap(), 1);
+        assert_eq!(hardage('G').unwrap(), 1);
+        assert_eq!(hardage('b').unwrap(), 1);
+        assert_eq!(hardage('z').unwrap(), 1);
+        assert_eq!(hardage('0').unwrap(), 2);
+        assert_eq!(hardage('1').unwrap(), 2);
+        assert_eq!(hardage('2').unwrap(), 2);
+        assert_eq!(hardage('3').unwrap(), 2);
+        assert_eq!(hardage('4').unwrap(), 2);
+    }
+
+    #[test]
+    #[should_panic(expected = "UnexpectedCode(\"count code start\")")]
+    fn test_unexpected_count_code() {
+        hardage('-').unwrap();
+    }
+
+    #[test]
+    #[should_panic(expected = "UnexpectedCode(\"op code start\")")]
+    fn test_unexpected_op_code() {
+        hardage('_').unwrap();
+    }
+
+    #[test]
+    #[should_panic(expected = "UnknownHardage(\"8\")")]
+    fn test_unknown_hardage() {
+        hardage('8').unwrap();
+    }
```

Comment:
This is another option:
```rust
assert!(hardage('8').is_err());
```

It wouldn't get the strings to make sure we are constructing them correctly, but it's easier to maintain.

Reply:
Yes I saw your other tests, it's on my list to update. Thanks